Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/DataSynchronizer.swift

```diff
@@ -1376,57 +1364,56 @@ public class DataSynchronizer: NetworkStateDelegate, FatalErrorListener {
      Starts data synchronization in a background thread.
      */
     public func start() {
-        syncLock.writeLock()
-        defer { syncLock.unlock(for: .writing) }
-        if (!self.isConfigured) {
-            return
-        }
+        try? syncLock.write {
```

Comment:
[minor] the `try?` seems smelly to me. makes it seem like we're just squashing any errors that happen. The old implementation caught any errors that happened within this block, so if you catch all the errors here then the `lock.write` shouldn't be re-throwing

if you don't want to solve it here, add a TODO for the error handling refactor ticket

Reply:
We can bubble up the errors here into the fatal error listener.

Reply:
Actually there are no errors here.